server/src/main/java/org/elasticsearch/ingest/BulkRequestPreprocessor.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.ingest;
+
+import org.elasticsearch.action.DocWriteRequest;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.cluster.metadata.Metadata;
+
+import java.util.concurrent.ExecutorService;
+import java.util.function.BiConsumer;
+import java.util.function.IntConsumer;
+
+public interface BulkRequestPreprocessor {
```

Comment:
A common interface for preprocessing a bulk request. Both IngestService and FieldInferenceBulkRequestPreprocessor implements it for using pipelines and model inference respectively.

Reply:
Seems like we don't need this. Both actions implement the abstract class. I think this interface should be removed.

Reply:
Makes sense - I'll remove that in a later iteration 👍